### PR TITLE
[Merged by Bors] - feat: y*log(x)=log(z) iff x^y=z

### PR DESCRIPTION
**Pull request by kbuzzard**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

Seems to be missing?

### CODE REVIEW

File: Mathlib/Analysis/SpecialFunctions/Pow/Real.lean [hunk 1]
```diff
@@ -403,6 +403,17 @@ theorem log_rpow {x : ℝ} (hx : 0 < x) (y : ℝ) : log (x ^ y) = y * log x := b
   rw [exp_log (rpow_pos_of_pos hx y), ← exp_log hx, mul_comm, rpow_def_of_pos (exp_pos (log x)) y]
 #align real.log_rpow Real.log_rpow
 
+theorem rpow_eq_of_mul_log {x y z : ℝ} (hx : 0 < x) (hz : 0 < z) (h : y * log x = log z) :
+    x ^ y = z := by
+  rw [← log_rpow hx] at h
+  apply_fun Real.exp at h
+  rwa [exp_log, exp_log hz] at h
+  exact rpow_pos_of_pos hx y
```
Comment (eric-wieser) [inline Mathlib/Analysis/SpecialFunctions/Pow/Real.lean:411]:
```suggestion
    x ^ y = z :=
  log_injOn_pos (rpow_pos_of_pos hx _) hz <| log_rpow hx _ |>.trans h
```

File: Mathlib/Analysis/SpecialFunctions/Pow/Real.lean [hunk 2]
```diff
@@ -403,6 +403,17 @@ theorem log_rpow {x : ℝ} (hx : 0 < x) (y : ℝ) : log (x ^ y) = y * log x := b
...
+
+theorem rpow_eq_iff_mul_log {x y z : ℝ} (hx : 0 < x) (hz : 0 < z) :
```
Comment (eric-wieser) [inline Mathlib/Analysis/SpecialFunctions/Pow/Real.lean:413]:
```suggestion
theorem mul_log_eq_log_iff {x y z : ℝ} (hx : 0 < x) (hz : 0 < z) :
```
matches the statement

**Comment by ADedecker** [comment]
Since the assumptions are the same why not just do the `iff` lemma? I don't think this direction will be useful enough to justify having it as a separate implication.

**Comment by kbuzzard** [comment]
OK I've removed the one-way implication. I'm now not sure about conventions: is
```
  ⟨...,
  by ...⟩
```
correct?

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Pull request closed without merging**